pkgs/development/python-modules/cupy/default.nix

```diff
@@ -0,0 +1,51 @@
+{ stdenv, python, buildPythonPackage
+, fetchPypi, isPy3k, linuxPackages, gcc5
+, fastrlock, numpy, six, wheel, pytest, mock
+, cudatoolkit, cudnn, nccl
+}:
+
+buildPythonPackage rec {
+  name = "${pname}-${version}";
+  pname = "cupy";
+  version = "2.2.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0si0ri8azxvxh3lpm4l4g60jf4nwzibi53yldbdbzb1svlqq060r";
+  };
+
+  checkInputs = [
+    pytest
+    mock
+  ];
+
+  nativeBuildInputs = [
+    gcc5
+  ];
+
+  propagatedBuildInputs = [
+    cudatoolkit
+    cudnn
+    linuxPackages.nvidia_x11
+    nccl
+    fastrlock
+    numpy
+    six
+    wheel
+  ];
+
+  CUDA_PATH = "${cudatoolkit}";
+  CFLAGS = "-I ${cudnn}/include -I ${nccl}/include";
+  LDFLAGS = "-L ${cudnn}/lib -L ${nccl}/lib";
```

Comment:
Are these three lines necessary? It builds fine for me without them.

Reply:
According to cupy's official install guide document, we need set environment values when cupy will be installed with cuda, cudnn, nccl.
https://docs-cupy.chainer.org/en/stable/install.html#install-cupy-with-cuda
https://docs-cupy.chainer.org/en/stable/install.html#install-cupy-with-cudnn-and-nccl

Reply:
If I understand the note correctly, it should not be necessary. Nix will add `${cudatoolkit}/bin/nvcc` to `PATH`, as well as set the appropriate `CFLAGS` and `LDFLAGS`.

Reply:
The official document prefers to use pip installing and set envs with 'export'.
I'm worrying those environment values are needed in python phase.
I think Nix will not add CFLAGS and LDFLAGS to python.

Actually, those checks are in python.
https://github.com/cupy/cupy/blob/06e84b561aeba2d4abc5342fe5a7c1da894a436b/install/build.py#L33
https://github.com/cupy/cupy/blob/df93794d685560ac6f44b23528e81eb501e7871c/cupy_setup_build.py#L250

Reply:
Nix sets those variables in python since it is based on [`mkDerivation`](https://github.com/NixOS/nixpkgs/blob/53c2210812aab64dec10d156720047adea5e5bec/pkgs/development/interpreters/python/mk-python-derivation.nix#L63).

Reply:
Sorry.
You are right, those settings are unnecessary.
I delete those.

I found out ```CUDA_PATH``` is unnecessary when ```PATH``` has ```${cudatoolkit}/bin```. 
And I attempted building after delete those environment variables settings, and 
chainer could find cudnn.

```
>>> import chainer
>>> chainer.cuda.available
True
>>> chainer.cuda.cudnn_enabled
True
>>> 
```